pallets/auctions/src/lib.rs

```diff
@@ -0,0 +1,223 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+pub use pallet::*;
+
+#[cfg(test)]
+mod mock;
+
+#[cfg(test)]
+mod tests;
+
+#[cfg(feature = "runtime-benchmarks")]
+mod benchmarking;
+
+mod types;
+
+#[frame_support::pallet]
+pub mod pallet {
+    use crate::types::AuctionData;
+    use frame_support::sp_runtime::traits::CheckedSub;
+    use frame_support::traits::Currency;
+    use frame_support::{dispatch::DispatchResultWithPostInfo, pallet_prelude::*};
+    use frame_system::pallet_prelude::*;
+    use ternoa_common::traits::NFTTrait;
+    use ternoa_primitives::{nfts::NFTId, MarketplaceId};
+
+    pub type BalanceCaps<T> =
+        <<T as Config>::CurrencyCaps as Currency<<T as frame_system::Config>::AccountId>>::Balance;
+
+    /// Configure the pallet by specifying the parameters and types on which it depends.
+    #[pallet::config]
+    pub trait Config: frame_system::Config {
+        /// Because this pallet emits events, it depends on the runtime's definition of an event.
+        type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
+        /// Caps Currency
+        type CurrencyCaps: Currency<Self::AccountId>;
+        /// Get information on nfts
+        type NFTHandler: NFTTrait<AccountId = Self::AccountId>;
+        /// Minimum required length of auction
+        #[pallet::constant]
+        type MinAuctionDuration: Get<Self::BlockNumber>;
+        /// Maximum permitted length of auction
+        #[pallet::constant]
+        type MaxAuctionDuration: Get<Self::BlockNumber>;
+        /// Minimum buffer of blocks required before auction start
+        #[pallet::constant]
+        type MinAuctionBuffer: Get<Self::BlockNumber>;
+    }
+
+    #[pallet::pallet]
+    #[pallet::generate_store(pub(super) trait Store)]
+    pub struct Pallet<T>(_);
```

Comment:
Don't forget to add `#[pallet::storage_version(STORAGE_VERSION)]`.

And `const STORAGE_VERSION: StorageVersion = StorageVersion::new(1);` too